src/spammer/spammer.ts

```diff
@@ -1,27 +1,46 @@
-import { GaxiosError, request } from "gaxios";
+import { GaxiosError, GaxiosResponse, request } from "gaxios";
 
 import { logComplete, logDownload, logError } from "../cli/logger";
 import { Config } from "../models/config.model";
+import { NpmjsResponse } from "../models/npmjs-response.model";
 
 const sleep = async (milliseconds: number): Promise<void> => {
     return new Promise((resolve) => setTimeout(resolve, milliseconds));
 };
 
-export const downloadPackage = async (name: string, version: string): Promise<void> => {
-    const packageUrl = `https://registry.yarnpkg.com/${name}/-/${name}-${version}.tgz`;
+export const queryNpms = async (name: string): Promise<NpmjsResponse> => {
+    let npmsResponse: GaxiosResponse<NpmjsResponse> | undefined;
+
+    await request<NpmjsResponse>({
+        baseUrl: "https://api.npms.io",
+        url: `/v2/package/${name}`,
+        method: "GET",
+    })
+        .then((response: GaxiosResponse<NpmjsResponse>) => (npmsResponse = response))
+        .catch((response: GaxiosError<NpmjsResponse>) => {
+            throw Error(`Failed to download ${response.config.url}\n${response.message}`);
+        });
 
+    return (npmsResponse as GaxiosResponse<NpmjsResponse>).data;
+};
```

Comment:
I would rather not type cast this.  Is there another way I can ensure that npmsResponse is defined?